File: 8_kyu/Beginner_Series_2_Clock.lua
-- Converts hours, minutes, seconds into milliseconds.
return function(hours, minutes, seconds)
  -- also accept nil as 0
  hours = hours or 0
  minutes = minutes or 0
  seconds = seconds or 0
  -- convert into milliseconds
  return ((hours*60 + minutes)*60 + seconds)*1000
end

File: 8_kyu/Beginner_Series_2_Clock.kt
// Factor to convert seconds in milliseconds.
const val SECOND_MS: Int = 1000

// Factor to convert minutes in seconds.
const val MINUTE_S: Int = 60

// Factor to convert houres in minutes.
const val HOURE_M: Int = 60

// Converts time h:m:s into milliseconds.
fun past(h: Int, m: Int, s: Int) = ((h*HOURE_M + m)*MINUTE_S + s)*SECOND_MS
